qemu/tests/netperf_udp_perf.py

```diff
@@ -0,0 +1,229 @@
+    host = params.get("host", "localhost")
+    host_ip = host
+    client = params.get("client", "localhost")
+    client_ip = client
+    client_pub_ip = params.get("client_pub_ip")
+    client = remote.wait_for_login(params.get("shell_client_client"),
+                                   client_pub_ip,
+                                   params.get("shell_port_client"),
+                                   params.get("username_client"),
+                                   params.get("password_client"),
+                                   params.get("shell_prompt_client"))
+    cmd = "ifconfig %s %s up" % (params.get("client_physical_nic"),
+                                 client_ip)
+    netperf_base.ssh_cmd(client, cmd)
+
+    error_context.context("Prepare env of server/client/host", logging.info)
+    prepare_list = set([server_ctl, client, host])
+    tag_dict = {server_ctl: "server", client: "client", host: "host"}
+    ip_dict = {server_ctl: server_ctl_ip, client: client_pub_ip,
+               host: host_ip}
+    for i in prepare_list:
+        params_tmp = params.object_params(tag_dict[i])
+        shell_port = int(params_tmp["shell_port"])
+        password = params_tmp["password"]
+        username = params_tmp["username"]
+        netperf_base.env_setup(test, params, i, ip_dict[i],
+                               username, shell_port, password)
+
+    netperf_base.tweak_tuned_profile(params, server_ctl, client, host)
+
+    env.stop_ip_sniffing()
+
+    try:
+        error_context.context("Start netperf udp stream testing", logging.info)
+        start_test(server_ip, server_ctl, host, client, test.resultsdir,
+                   test_duration=int(params.get('l')),
+                   burst_time=params.get('burst_time'),
+                   numbers_per_burst=params.get('numbers_per_burst'),
+                   ver_cmd=params.get('ver_cmd', "rpm -q qemu-kvm"),
+                   params=params, test=test)
+
+        if params.get("log_hostinfo_script"):
+            src = os.path.join(test.virtdir, params.get("log_hostinfo_script"))
+            path = os.path.join(test.resultsdir, "systeminfo")
+            process.system_output("bash %s %s &> %s" % (
+                                  src, test.resultsdir, path), shell=True)
+
+        if params.get("log_guestinfo_script") and params.get("log_guestinfo_exec"):
+            src = os.path.join(test.virtdir, params.get("log_guestinfo_script"))
+            path = os.path.join(test.resultsdir, "systeminfo")
+            destpath = params.get("log_guestinfo_path", "/tmp/log_guestinfo.sh")
+            vm.copy_files_to(src, destpath, nic_index=1)
+            logexec = params.get("log_guestinfo_exec", "bash")
+            output = server_ctl.cmd_output("%s %s" % (logexec, destpath))
+            logfile = open(path, "a+")
+            logfile.write(output)
+            logfile.close()
+    except process.CmdError as err:
```

Comment:
`err` is unused? Then `except process.CmdError` would be fine. But I would recommend to raise it for debug usage, depends on your test requirement.

Reply:
yes, no need err, I have updated.